src/composables/useRoutesMap.ts

```diff
@@ -0,0 +1,227 @@
+      centerMapOnExtent(extent);
+    }
+  };
+
+  /**
+   * Get the extent of a route.
+   * If route is not a Feature, returns null.
+   * @param {Feature | Ref<Feature>} route - The LineString route.
+   * @return {Extent | null} The extent of the route.
+   */
+  const getRouteExtent = (route: Feature | Ref<Feature>): Extent | null => {
+    const geometry = unref(route).getGeometry();
+    return geometry ? geometry.getExtent() : null;
+  };
+
+  /**
+   * Centers the map on the given extent.
+   * Takes the MAX_ZOOM_CENTERING_FACTOR into account.
+   * @param {Extent} extent - The extent to center the map on.
+   * @return {void}
+   */
+  const centerMapOnExtent = (extent: Extent): void => {
+    const map = mapRef.value?.map;
+    if (map) {
+      const view = map.getView();
+      view.fit(extent, {
+        size: map.getSize(),
+        maxZoom: MAX_ZOOM_CENTERING_FACTOR,
+      });
+    }
+  };
+
+  /**
+   * Get the length of a route based on the geometry.
+   * @param {Feature | Ref<Feature>} route - The LineString route.
+   * @return {number} The length of the route.
+   */
+  const getRouteLength = (route: Feature | Ref<Feature>): number => {
+    const geom = unref(route).getGeometry();
+    if (geom instanceof LineString) {
+      return getLength(geom);
+    }
+    return 0;
+  };
+
+  /**
+   * Returns a string representing the length of a route in kilometers.
+   * @param {FeatureRoute} featureRoute - The route to calculate the length for.
+   * @return {string} A string representing the length of the route in kilometers.
+   */
+  const getRouteLengthLabel = (featureRoute: FeatureRoute): string => {
+    const length = getRouteLength(featureRoute.route);
+    return formatLength(length);
+  };
+
+  /**
+   * Formats the length value to a rounded string with the global route length unit.
+   * @param {number} length - The length value to format.
+   * @return {string} The formatted length string.
+   */
+  const formatLength = (length: number) => {
+    const lengthRoudned = Math.round(length * 100);
+    return `${lengthRoudned} ${i18n.global.t('global.routeLengthUnit')}`;
+  };
+
+  onMounted(() => {
+    useGeographic();
+  });
+
+  /**
+   * Centers the map on the current location if geolocation is supported.
+   * If geolocation is not supported, an error message is logged to the console.
+   * @return {void}
+   */
+  const centerOnCurrentLocation = (): void => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          const coords = fromLonLat([
+            position.coords.longitude,
+            position.coords.latitude,
+          ]);
+          center.value = coords;
+          if (mapRef.value?.map) {
+            const view = mapRef.value.map.getView();
+            view.setCenter(coords);
+            view.setZoom(DEFAULT_MAP_ZOOM);
+          }
+        },
+        (error) => {
+          console.error('Geolocation error:', error);
```

Comment:
User don't know about web dev console. Use modal dialog to show error message, please.

Reply:
Replaced console log with Quasar Notify api https://quasar.dev/quasar-plugins/notify#notify-api